libs/types/src/tokens.rs

```diff
@@ -210,6 +199,61 @@ pub struct CustomMetadata {
 
 	/// Whether an asset can be used as a currency to fund Centrifuge Pools.
 	pub pool_currency: bool,
+
+	/// The ways, if any, this token is cross-chain transferable
+	pub transferability: Option<CrossChainTransferability>,
```

Comment:
That doesn't work since it wouldn't allow for that value / XCM metadata to be set for CrossChai Transferability::All, unless if we would also have XcmMetadata as a value for that variant, but that seems a bit odd.

Reply:
Wdyt?

Reply:
It actually works, happy to push my commit which moves `XcmMetadata` into `CrossChainTransferability::XCM`, e.g. ... with fixes

```rust
pub enum CrossChainTransferability {
	/// The asset is only transferable through XCM
	Xcm(XcmMetadata),

	/// The asset is only transferable through Centrifuge Connectors
	Connectors,

	/// The asset is transferable through all available options
	All,
}

impl Default for CrossChainTransferability {
	fn default() -> Self {
		Self::Xcm(Default::default())
	}
}
```

Reply:
@wischli The issue that I am describing above with this approach is that if we set a token to `CrossChainTransferability::All`, which presumes that the asset at hand is also XCM transferable, we won't have a `XcmMetadata` to retrieve anymore; we will only have that if we _also_ have `XcmMetadata` as a field of `CrossChainTransferability::All`:

``` rust
pub enum CrossChainTransferability {
	/// The asset is only transferable through XCM
	Xcm(XcmMetadata),

	/// The asset is only transferable through Centrifuge Connectors
	Connectors,

	/// The asset is transferable through all available options
	All(XcmMetadata), // HERE
}
```

I said above that this alternative seems a bit awkward but coming back at this today I can't really find a good argument as to why it's awkward so if it sounds good to you both, it sounds good to me :)

Reply:
Ah good point, I was blinding that out. I would still prefer this over having the xcm metadata separated. Going this path, we also ensure we need to run migrations if we add more variants to `CrossChainTransferability` with additional metadata that needs to be added to `All`.

Reply:
Not really. It's true I am dependent on this PR but I won't finish before anyways.